examples/failing/ConstraintInference.purs

```diff
@@ -0,0 +1,10 @@
+-- @shouldFailWith NoInstanceFound
+
+module Main where
+
+import Prelude
+
+spin :: forall a b. a -> b
+spin x = spin x
```

Comment:
Because it's trying to find an instance for an unknown, but an unknown which can't be generalized. I considered adding another error message type here, but every message I came up with was very confusing, so it was easier to just reuse this one.